saleor/tests/e2e/checkout/test_unlogged_customer_is_unable_to_buy_unpublished_product.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+
+from ..channel.utils import create_channel
+from ..product.utils import (
+    create_category,
+    create_product,
+    create_product_channel_listing,
+    create_product_type,
+    create_product_variant,
+    create_product_variant_channel_listing,
+    raw_create_product_channel_listing,
+)
+from ..shipping_zone.utils import (
+    create_shipping_method,
+    create_shipping_method_channel_listing,
+    create_shipping_zone,
+)
+from ..utils import assign_permissions
+from ..warehouse.utils import create_warehouse
+from .utils import raw_checkout_create
+
+
+def prepare_product(
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+    permission_manage_apps,
+    permission_handle_checkouts,
+    channel_slug,
+):
+    permissions = [
+        permission_manage_products,
+        permission_manage_channels,
+        permission_manage_shipping,
+        permission_manage_product_types_and_attributes,
+        permission_manage_apps,
+        permission_handle_checkouts,
```

Comment:
This permission is not needed for this test. HANDLE_CHECKOUTS is needed for applications for use the orderCreateFromCheckout mutation